src/parser/core/Entity.ts

```diff
@@ -147,6 +148,67 @@ class Entity {
     return Object.keys(stackBuffUptimes).map(stack => stackBuffUptimes[Number(stack)] * Number(stack)).reduce((total, cur) => total + cur, 0);
   }
 
+  /**
+   * Determine the non-inclusive remaining time of a given debuff/buff at each application/re-application of the buff.
+   * @param {number} spellId - buff ID to check for
+   * @param {number} baseBuffLength - the base number of ms added to the buff timer by a single application
+   * @param {number} maxBuffLength - the maximum number of ms a buff can have.
+   * @param {number | null} sourceId - source ID the buff must have come from, or any source if null.
+   */
+  getRemaingBuffTimeAtApplication(spellId: number, baseBuffLength: number, maxBuffLength: number, sourceId: number | null = null): Map<number, number> {
```

Comment:
tbh i don't understand why we are moving this. It feels like there is no real benefit.

Reply:
To clarify on the comment in the summary: This is fresh code that I originally discussed adding in Discord and it was suggested as putting it in the `Buffs` analyzer. I originally placed it there and did the remainder of the rogue work, then refactored it into `Entity.ts` when I found that I was simply operating on an entity parameter within `Buffs` anyways. We're not moving it between files, I'm just specifying for why I went against the original suggestion in Discord.